happi/cli.py

```diff
@@ -94,53 +96,68 @@ def search(ctx, show_json, names, search_criteria):
                 'Received duplicate search criteria %s=%r (was %r)',
                 criteria, value, client_args[criteria]
             )
-            return
-        if value.replace('.', '').isnumeric():
-            logger.debug('Changed %s to float', value)
-            value = str(float(value))
+            raise click.Abort()
 
         if is_a_range(value):
             start, stop = value.split(',')
             start = float(start)
             stop = float(stop)
-            is_range = True
             if start < stop:
-                range_list = client.search_range(criteria, start, stop)
+                new_range_list = client.search_range(criteria, start, stop)
             else:
                 logger.error('Invalid range, make sure start < stop')
+                raise click.Abort()
+
+            if not range_found:
+                # if first range, just replace
+                range_found = True
+                range_list = new_range_list
+            else:
+                # subsequent ranges, only take intersection
+                final_range = [item for item in new_range_list
+                               if item in range_list]
+                range_list = final_range
```

Comment:
This is totally good and probably doesn't need to change, but you should know that the `set` type in python makes this trivial via the `intersection` method. There are probably also other parts of this implementation that are simplified by using sets instead of lists.
https://docs.python.org/3.10/library/stdtypes.html#frozenset.intersection

Reply:
I thought about using sets, but the SearchResult items aren't hashable so we'd have to grab the id's, do the set operations, and go back to the search results.  

I figured that was more effort than the list comprehension, and from what I can tell it's not too much slower

Reply:
Hmm, we could make `SearchResult` hashable - I think that'd be a good improvement regardless of the implementation here